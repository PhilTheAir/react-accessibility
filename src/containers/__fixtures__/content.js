import React from 'react';
import * as Ionicons from 'react-icons/lib/io';

export const links = [
  {
    "title": "Android",
    "href": "https://www.android.com",
    "icon": <Ionicons.IoSocialAndroid />
  },
  {
    "title": "Node.js",
    "href": "https://nodejs.org/en/",
    "icon": <Ionicons.IoSocialNodejs />
  },
  {
    "title": "GitHub",
    "href": "https://github.com",
    "icon": <Ionicons.IoSocialGithub />
  }
]