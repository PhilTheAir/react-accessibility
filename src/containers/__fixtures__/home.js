import React from 'react';
import * as Ionicons from 'react-icons/lib/io';

export const links = [
  {
    "title": "Home",
    "href": "/",
    "icon": <Ionicons.IoHome />
  },
  {
    "title": "Page Structures",
    "href": "/page",
    "icon": <Ionicons.IoIosStarOutline />
  },
  {
    "title": "Menus",
    "href": "/menus",
    "icon": <Ionicons.IoAndroidMenu />
  },
  {
    "title": "Images",
    "href": "/images",
    "icon": <Ionicons.IoImages />
  },
  {
    "title": "Tables",
    "href": "/tables",
    "icon": <Ionicons.IoIosTennisball />
  },
  {
    "title": "Forms",
    "href": "/forms",
    "icon": <Ionicons.IoIosFolder />
  }
];

export const lists = [
  {
    "id": "i0",
    "name": "React"
  },
  {
    "id": "i1",
    "name": "Redux"
  },
  {
    "id": "i2",
    "name": "Angular"
  },
  {
    "id": "i3",
    "name": "Vue"
  },
  {
    "id": "i4",
    "name": "Lodash"
  },
]

export const notes = {
  "title": "Author",
  "name": "Liwen CHEN",
}