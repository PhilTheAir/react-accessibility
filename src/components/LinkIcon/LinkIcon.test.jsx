import React from 'react';
import { LinkIcon } from './index';
import { shallow } from 'enzyme';
import { links } from '../../containers/__fixtures__/content';

describe('LinkIcon', () => {
  it('should render', () => {
    const link = links[0];
    const component = <LinkIcon
      key={link.title}
      title={link.title}
      link={link.href}
      icon={link.icon}
    />;
    const wrapper = shallow(component).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
