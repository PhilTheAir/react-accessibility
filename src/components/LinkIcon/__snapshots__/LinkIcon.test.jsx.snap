// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkIcon should render 1`] = `
.c0 {
  font-size: 64px;
  margin: 24px;
  border: none;
  background: none;
}

.c0:hover {
  cursor: pointer;
  color: pink;
}

<SubscribedComponent
  aria-label="Android"
  className="c0"
  is="button"
  onClick={[Function]}
  role="link"
  tabIndex={0}
  title="Android"
>
  <IoSocialAndroid />
</SubscribedComponent>
`;
